shared/src/main/java/io/pravega/shared/segment/StreamSegmentNameUtils.java

```diff
@@ -50,7 +50,7 @@
     /**
      * This is appended to the end of the Primary Segment Name, followed by epoch.
      */
-    private static final String EPOCH_DELIMITER = "#epoch.";
+    private static final String EPOCH_DELIMITER = ".#epoch.";
```

Comment:
As I have asked before, it seems unnecessary to make this change.

Reply:
this change is done to cater to needs mentioned in #2681 
We added this delimiter in previous change (#2469) and there was an ask to add an extra "." to help with metrics name parsing.